### PR TITLE
[FIX #51] Allow to use different streams for Redis adapter

### LINKED ISSUE #51: Question: Allow to use different streams for redis adapter?

### PR DESCRIPTION
**Pull request by smakagon**

### CODE REVIEW

File: lib/hanami/events/adapter/redis.rb [hunk 1]
```diff
@@ -15,11 +15,12 @@ def initialize(params)
           @logger = params[:logger]
           @subscribers = []
           @thread_spawned = false
+          @stream = params[:stream] || STREAM_NAME
         end
 
         def broadcast(event_name, payload)
           @redis.with do |conn|
-            conn.lpush(STREAM_NAME, {
+            conn.lpush(@stream, {
```
Comment (davydovanton) [inline lib/hanami/events/adapter/redis.rb]: WDYT, maybe is better to create getter `stream` instead instance variable? I mean:
```ruby
def stream
  @stream || DEFAULT_STREAM
end
```

Comment (smakagon) [inline lib/hanami/events/adapter/redis.rb]: 👍 good idea

**Comment by smakagon** [comment]
@davydovanton I've added specs and made changes according to review.

File: lib/hanami/events/adapter/redis.rb [hunk 2]
```diff
@@ -15,11 +15,12 @@ def initialize(params)
           @logger = params[:logger]
           @subscribers = []
           @thread_spawned = false
+          @stream = params[:stream]
```
Comment (artofhuman) [inline lib/hanami/events/adapter/redis.rb]: may be better to use fetch here instead use method ?
``` ruby
@stream = params.fetch(:stream, DEFAULT_STREAM)
```

Comment (smakagon) [inline lib/hanami/events/adapter/redis.rb]: @artofhuman thanks for the idea. I like it.

File: lib/hanami/events/adapter/redis.rb (hunk 1 above)
Comment (smakagon) [inline lib/hanami/events/adapter/redis.rb]: I've rewritten `initialize` to this:
`params.fetch(:stream, DEFAULT_STREAM)` which feels good enough for this case. 
I like to see the default value at initialization point.

**Comment by davydovanton** [comment]
great work!

**Pull request merged**